ivi-layermanagement-examples/simple-weston-signal/src/simple-weston-signal.c

```diff
@@ -0,0 +1,137 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <signal.h>
+
+#include <weston-simple-client-protocol.h>
+
+static int running = 1;
+
+typedef struct _WaylandContext {
+        struct wl_display       *wl_display;
+        struct wl_registry      *wl_registry;
+        struct wl_compositor    *wl_compositor;
+        struct weston_simple    *weston_simple;
+}WaylandContextStruct;
+
+static void
+signal_format(void *data,
+	      struct weston_simple *weston_simple,
+	      uint32_t signal_response)
+{
+        if(signal_response == 1)
+        {
+                printf("Server received a signal from client!\n");
+        }
+        else
+        {
+                printf("Server did not receive a signal from client!\n");
+        }
+}
+
+static struct weston_simple_listener signal_listenter = {
+        signal_format
+};
+
+static void
+registry_handle_global(void *data, struct wl_registry *registry, uint32_t name,
+                       const char *interface, uint32_t version)
+{
+        WaylandContextStruct* wlcontext = (WaylandContextStruct*)data;
+
+        if (!strcmp(interface, "wl_compositor")) {
+                wlcontext->wl_compositor =
+                        wl_registry_bind(registry, name,
+                                         &wl_compositor_interface, 1);
+        }
+        else if (!strcmp(interface, "weston_simple")) {
+                wlcontext->weston_simple =
+                        wl_registry_bind(registry, name,
+                                         &weston_simple_interface, 1);
+                weston_simple_add_listener(wlcontext->weston_simple, &signal_listenter, wlcontext);
+        }
+}
+
+static void
+registry_handle_global_remove(void *data, struct wl_registry *registry,
+                              uint32_t name)
+{
+
+}
+
+static const struct wl_registry_listener registry_listener = {
```

Comment:
not consistent with above function

Reply:
sorry, this comment is in the wrong place, with the struct you can leave it as current, but with function, see:

```
static void
registry_handle_global_remove()
```

not consistent with:

`int init_wayland_context()`

Reply:
> `int init_wayland_context()`
Yes, I updated it.